src/components/multi-select/multi-select.test.jsx

```diff
@@ -0,0 +1,516 @@
+import React, { createRef } from 'react';
+import {
+  render,
+  screen,
+  waitFor,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import MultiSelect from './multi-select.jsx';
+import Tag from '../tag/tag.jsx';
+import {
+  findAutocompleter,
+  findAvailableOption,
+  findRemoveButton,
+  findSelectedOption,
+  openOptions,
+  queryAvailableOption,
+  querySelectedOption,
+  queryRemoveButton,
+} from './test-queries.js';
+
+describe('<MultiSelect>', () => {
+  const requiredProps = {
+    options: ['Foo', 'Bar'],
+    id: 'test-id',
+    label: 'test label',
+    name: 'field-id',
+  };
+
+  it('has defaults', () => {
+    const ref = createRef();
+
+    render(<MultiSelect {...requiredProps} ref={ref} />);
+    expect(document.body).toMatchSnapshot();
+    expect(ref.current).toMatchSnapshot();
+  });
+
+  describe('classNames API', () => {
+    it('uses class name overrides provided', () => {
+      render(<MultiSelect
+        {...requiredProps}
+        classNames={{
+          container: 'unique-container',
+          formControlChildrenContainer: 'unique-form-control-children-container',
+          input: 'unique-input-class',
+          tagList: 'unique-tag-list-class',
+        }}
+      />);
+
+      expect(document.querySelector('.unique-container')).toBeInTheDocument();
```

Comment:
Welcome to use a snapshot for these kinds of tests too!

Reply:
Ooooh, that would make it more straightforward. I'll do that!